isolated-contexts.bs

```diff
@@ -44,6 +570,104 @@ If the [{{IsolatedContext}}] [=extended attribute=] appears on an
+[{{IsolatedContext}}] [=extended attribute=] does not further restrict the
+exposure of the [=member=].
+
+An [=interface=] without the [{{IsolatedContext}}] [=extended attribute=]
+must not [=interface/inherit=] from another interface
+that does specify [{{IsolatedContext}}].
+
+### Patches to the "exposed" algorithm ### {#monkey-webidl-exposed}
+
+WebIDL's [=exposed=] algorithm is adjusted as follows, adding a single step
+after similarly handling [{{CrossOriginIsolated}}] (step 4 below).
+
+<div algorithm>
+  An [=interface=], [=callback interface=], [=namespace=], or [=member=]
+  |construct| is <dfn id="dfn-exposed" export>exposed</dfn> in a given
+  [=realm=] |realm| if the following steps return true:
+
+  <ol>
+    <li>
+      If |construct|'s [=exposure set=] is not <code>*</code>, and
+      |realm|.\[[GlobalObject]] does not implement an [=interface=] that is in
+      |construct|'s [=exposure set=], then return false.
+    </li>
+    <li>
+      If |realm|'s [=realm/settings object=] is not a [=secure context=], and
+      |construct| is [=conditionally exposed=] on [{{SecureContext}}], then
+      return false.
+    </li>
+    <li>
+      If |realm|'s [=realm/settings object=]'s
+      [=environment settings object/cross-origin isolated capability=] is false,
+      and |construct| is [=conditionally exposed=] on [{{CrossOriginIsolated}}],
+      then return false.
+    </li>
+    <li><ins>
+        If |realm|'s [=realm/settings object=] does not
+        [=environment settings object/enforce isolation and integrity=], and
+        |construct| is [=conditionally exposed=] on [{{IsolatedContext}}], then
+        return `false`.
+    </ins></li>
+    <li>Return true.</li>
+  </ol>
+</div>
+
+## Storage ## {#monkey-storage}
+
+The [=obtain a storage key for non-storage purposes=] algorithm is extended to
+require double-keying on all storage within a [=browsing context group=]
+containing [=Isolated Contexts=].
+
+<div algorithm="obtain a storage key for non-storage purposes isolated context">
+To obtain a storage key for non-storage purposes, given an
+<a href="https://html.spec.whatwg.org/multipage/webappapis.html#environment">
+environment</a> |environment|, run these steps:
+
+<ol>
+    <li>
+      Let |origin| be |environment|'s [=environment settings object/origin=] if
+      |environment| is an [=environment settings object=]; otherwise
+      |environment|'s [=creation URL=]'s [=origin=].
+    </li>
+
+    <li><ins>
+      Let |integrity origin| be the [=browsing context group/integrity origin=]
+      of the [=browsing context group=] that |environment| belongs to.
+    </ins></li>
+
+    <li><ins>
+      If |integrity origin| is non-null, return a [=tuple=] consisting of
+      |integrity origin| and |origin|.
+    </ins></li>
+
+    <li>
+      Return a [=tuple=] consisting of |origin|.
+    </li>
+</ol>
+</div>
+
+Note: This is essentially a poorly specified version of
```

Comment:
Instead of "poorly" I recommend "minimally" and dash between the 2 as "minimally-specified".